pkgs/development/libraries/glibc/nix-nss-open-files.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/nss/nss_files/files-XXX.c b/nss/nss_files/files-XXX.c
+index 1db9e46127..6cef6b2f4e 100644
+--- a/nss/nss_files/files-XXX.c
++++ b/nss/nss_files/files-XXX.c
+@@ -40,6 +40,7 @@
+ #define ENTNAME_r	CONCAT(ENTNAME,_r)
+ 
+ #define DATAFILE	"/etc/" DATABASE
++#define NIX_DATAFILE "NIX_etc_" DATABASE
+ 
+ #ifdef NEED_H_ERRNO
+ # include <netdb.h>
+@@ -75,10 +76,20 @@ internal_setent (FILE **stream)
+ 
+   if (*stream == NULL)
+     {
+-      *stream = __nss_files_fopen (DATAFILE);
+-
++      const char *file = DATAFILE;
++
++      // use the Nix environment vaiable such as `NIX_etc_protocols`
++      char *path = getenv (NIX_DATAFILE);
```

Comment:
This should be definitely `secure_getenv` Just imagine someone is redirecting `/etc/passwd` in sudo...

Reply:
A nice example.  I wonder if overriding locale archive could also be a risk.  Perhaps we'd better switch it to `secure_getenv`, too.

Reply:
I was wondering why the locale patch doesn't have it.